src/AbstractClient.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JTL\GoPrometrics\Client;
+
+use GuzzleHttp\ClientInterface;
+
+abstract class AbstractClient
+{
+    private ?ClientInterface $client;
+    protected string $baseUrl = '';
+
+    public function __construct(
+        ?ClientInterface $client = null,
```

Comment:
Why is the ClientInterface nullable?

The Client is required in this class. w/o a client the whole instance does make sense.
And if there is a good reason why it should nullable so please decide if nullable or optimal.

Even the baseUrl should not be empty

Reply:
I did this for @drieschel and not for me. Why he need it ......